src/exceptions/api.error.js

```diff
@@ -0,0 +1,33 @@
+export class ApiError extends Error {
+  constructor({ message, status, errors }) {
+    super(message);
+
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    })
+  }
+
+  static unauthorized(errors) {
+    return new ApiError({
+      message: 'unauthorized',
+      errors,
+      status: 400,
```

Comment:
The status code for an unauthorized error should be 401, not 400. Change `status: 400` to `status: 401` to correctly represent the unauthorized HTTP status.